File: fizzbuzz.kt
@Suppress("UNUSED_PARAMETER")
fun main(args: Array<String>) {
    for (i in 1..100) {
        var p = true
        if (i%3 == 0) {
            print("Fizz")
            p = false
        }
        if (i%5 == 0) {
            print("Buzz")
            p = false
        }
        if (p) {
            print(i)
        }
        println("")
   }
}

File: fizzbuzz.pl
fizzBuzz(N, 'FizzBuzz') :- fizzBuzz(N, 'Fizz'), fizzBuzz(N, 'Buzz').
fizzBuzz(N, 'Fizz')     :- 0 =:= mod(N, 3).
fizzBuzz(N, 'Buzz')     :- 0 =:= mod(N, 5).
fizzBuzz(N, N).

writeList(K,K) :- !.
writeList(K,L) :- fizzBuzz(K, FB),
                  write(FB), write('\n'),
                  N is K+1,
                  writeList(N, L).

main :- writeList(1,101), halt.
